dev-lang/gnucobol: rename from dev-lang/open-cobol

XVilka opened the pull request:
Because mainstream changed the project name starting
from the 2.0 versions, along with license (GPL3/LGPL3).
Also bumps to the latest stable 2.2 version.

Package-Manager: Portage-2.3.66, Repoman-2.3.12
Closes: https://bugs.gentoo.org/641888

XVilka commented:
Ping?

XVilka commented:
Anyone alive here?

leio commented:
Missing pkgmove entry to have users seamlessly update to the new name. Is the removal of the old version immediately intentional?

XVilka commented:
@leio, I wasn't sure, I can keep the old version if required. Will update the PR.

leio commented:
I don't know, that's for the maintainer to decide, I suppose. I think if removed, it should be a separate commit, though easier to do it in the same, as you wouldn't need to adjust the old ebuild to work under the new name.

XVilka commented:
Ok, canceled removal updated the ebuild and added also gnucobol-3.0-rc1 version.

XVilka commented:
Ping?

mgorny commented:
I'm sorry but it seems that the current maintainer doesn't care about this package. Are you still around? If so, would you be interested in becoming a new maintainer? If so, please replace the current maintainer with yourself in metadata.xml as found in https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide#Proxied_maintainer_in_metadata.xml

XVilka commented:
Yep, will do in a couple days. Thanks!

On Sat, Nov 9, 2019, 5:47 AM Michał Górny <notifications@github.com> wrote:

> I'm sorry but it seems that the current maintainer doesn't care about this
> package. Are you still around? If so, would you be interested in becoming a
> new maintainer? If so, please replace the current maintainer with yourself
> in metadata.xml as found in
> https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide#Proxied_maintainer_in_metadata.xml
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gentoo/gentoo/pull/12067?email_source=notifications&email_token=AABRT7IJHAVYPKOLANIKKB3QSXM5ZA5CNFSM4HONBPRKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEDTOAGI#issuecomment-552001561>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABRT7NXJXVVKO5I7DV4DHLQSXM5ZANCNFSM4HONBPRA>
> .
>

mgorny commented on dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:
```diff
@@ -0,0 +1,34 @@
+EAPI=7
+
+MY_PV=$(ver_cut 1-2)
+MY_P="${PN}-${MY_PV}-rc1"
```
`MY_P=${P/_/-}` is how we usually deal with it. Especially, don't hardcode `rc1`.

mgorny commented on dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:
```diff
@@ -0,0 +1,34 @@
+HOMEPAGE="https://open-cobol.sourceforge.io"
+SRC_URI="mirror://sourceforge/open-cobol/${MY_P}.tar.gz"
+
+LICENSE="GPL-3 LGPL-3"
```
The sources say it's 'or newer', i.e. `GPL-3+ LGPL-3+`.

mgorny commented on dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:
```diff
@@ -0,0 +1,34 @@
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE="berkdb nls readline"
+
+RDEPEND="dev-libs/gmp:0
```
Prefer sorting unconditional deps first, then all conditionals. It makes it easier to verify them later.

flappyports commented:
@XVilka, package has been converted to EAPI=7 and future proofed for pkgmove if you so choose to keep working this PR.

XVilka commented:
Yes, I will, sorry for the delay. Will update the PR later today.

On Sun, Nov 24, 2019, 3:53 AM Aaron <notifications@github.com> wrote:

> @XVilka <https://github.com/XVilka>, package has been converted to EAPI=7
> and future proofed for pkgmove if you so choose to keep working this PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gentoo/gentoo/pull/12067?email_source=notifications&email_token=AABRT7PQUIVSHB77CGTGHNDQVGC4JA5CNFSM4HONBPRKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEE74O5Y#issuecomment-557827959>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABRT7LSREW2NOILCLZNQXTQVGC4JANCNFSM4HONBPRA>
> .
>

XVilka commented:
Addressed everything. Please recheck.

XVilka commented:
Anything else that I need to do? Lets merge it finally if possible @leio @mgorny

XVilka commented:
Ping? Anything else I should do?

XVilka commented:
Ping?

XVilka commented:
Updated the pkgmove to 1Q-2020, please review.

XVilka commented:
Ping?

XVilka commented:
With the renewed interest in COBOL (it's even [trending on Twitter](https://news.ycombinator.com/item?id=22784248)) after the famous [New Jersey call for COBOL developers](https://www.cnn.com/2020/04/08/business/coronavirus-cobol-programmers-new-jersey-trnd/index.html) to fix their systems and many people trying the free solutions like GnuCOBOL, please merge it as soon as possible - so these people will get the most relevant experience (a lot has changed since the rename!). This is the last call, thanks. Otherwise I just close this.

thesamesam commented on dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:
```diff
@@ -0,0 +1,33 @@
+# Copyright 1999-2019 Gentoo Authors
```
Date

thesamesam commented on dev-lang/gnucobol/gnucobol-2.2.ebuild:
```diff
@@ -0,0 +1,29 @@
+# Copyright 1999-2019 Gentoo Authors
```
Date

thesamesam commented on dev-lang/gnucobol/gnucobol-2.2.ebuild:
```diff
@@ -0,0 +1,29 @@
+LICENSE="GPL-3+ LGPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE="berkdb nls readline"
```
Where does readline get used?

thesamesam commented on dev-lang/gnucobol/gnucobol-2.2.ebuild:
```diff
@@ -0,0 +1,29 @@
+	sys-libs/ncurses:0
+	berkdb? ( sys-libs/db:4.8 )"
+DEPEND="${RDEPEND}
+	sys-devel/libtool"
```
Quoting from https://sourceforge.net/projects/open-cobol/files/: 
   ***
    NOTE - libltdl is NOT needed when installing on Linux,
    SUN Solaris, MAC OS, CentOS or Windows
        (including Cygwin, MingW and native windows).
    It is also NOT needed with later versions of AIX and HP-UX.
    (AIX >= 5.1 and HP-UX >= 11.1 are known to NOT require this).
    (Check if you have the "dlopen" function).
    ***

So is libtool needed or not?

thesamesam commented on dev-lang/gnucobol/gnucobol-2.2.ebuild:
```diff
@@ -0,0 +1,29 @@
+	sys-devel/libtool"
+
+PATCHES=(
+	"${FILESDIR}/${P}-disable-data-hooks.patch"
```
${PN}

thesamesam commented:
I've spoken to the maintainer (Patrick) and he has given proxy-maintainers permission to move forward with this.

Sorry for the delay, it is not the biggest of projects, and where a package has a maintainer listed, it is not usually priority and can get lost.

I have given a few comments here and it is on the list for the project to look at. Thanks for your patience.

thesamesam commented on dev-lang/gnucobol/gnucobol-2.2.ebuild:
```diff
@@ -0,0 +1,29 @@
+
+LICENSE="GPL-3+ LGPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
```
I am suspicious about retaining ppc [keywords](https://devmanual.gentoo.org/keywording/index.html) given how major of a change this is.

thesamesam commented on dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:
```diff
@@ -0,0 +1,33 @@
+
+RDEPEND="dev-libs/gmp:0
+	sys-libs/ncurses:0
+	berkdb? ( sys-libs/db:4.8 )"
```
Conditional runtime dependencies aren't allowed, and it looks like this is used at build-time anyway, so BDEPEND or (likely here) DEPEND.

XVilka replied:
I took these lines from the open-cobol ebuild: https://github.com/gentoo/gentoo/blob/master/dev-lang/open-cobol/open-cobol-1.1.ebuild#L20

thesamesam replied:
Don't trust past ebuilds ;)

thesamesam commented on dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:
```diff
@@ -0,0 +1,33 @@
+	sys-libs/ncurses:0
+	berkdb? ( sys-libs/db:4.8 )"
+DEPEND="${RDEPEND}
+	sys-devel/libtool"
```
See comment about libtool above.

thesamesam commented on dev-lang/gnucobol/gnucobol-2.2.ebuild:
```diff
@@ -0,0 +1,28 @@
+LICENSE="GPL-3+ LGPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="berkdb nls"
```
We may need to pull in libntl or gettext if nls is enabled?

XVilka replied:
Added gettext - looks it is used by the source code.

thesamesam replied:
In the past, I have done:
BDEPEND="sys-devel/gettext"
DEPEND="virtual/libintl" (to allow for Musl where we need to pull in a package)

But I had to do some investigation to figure out what was needed and when. If you can't figure it out for now, we will leave it.

XVilka commented:
Let's leave it, they are going to prepare release in a few months, then it
can be reevaluated.

On Mon, May 4, 2020, 21:43 sam_c <notifications@github.com> wrote:

> *@thesamesam* commented on this pull request.
> ------------------------------
>
> In dev-lang/gnucobol/gnucobol-2.2.ebuild
> <https://github.com/gentoo/gentoo/pull/12067#discussion_r419445887>:
>
> > @@ -0,0 +1,28 @@
> +# Copyright 1999-2020 Gentoo Authors
> +# Distributed under the terms of the GNU General Public License v2
> +
> +EAPI=7
> +
> +DESCRIPTION="an open-source COBOL compiler"
> +HOMEPAGE="https://open-cobol.sourceforge.io"
> +SRC_URI="mirror://sourceforge/open-cobol/${P}.tar.gz"
> +
> +LICENSE="GPL-3+ LGPL-3+"
> +SLOT="0"
> +KEYWORDS="~amd64 ~x86"
> +IUSE="berkdb nls"
>
> In the past, I have done:
> BDEPEND="sys-devel/gettext"
> DEPEND="virtual/libintl" (to allow for Musl where we need to pull in a
> package)
>
> But I had to do some investigation to figure out what was needed and when.
> If you can't figure it out for now, we will leave it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gentoo/gentoo/pull/12067#discussion_r419445887>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABRT7JKESQKMG6QZV2QCNTRP3BAXANCNFSM4HONBPRA>
> .
>

XVilka commented:
So? Isn't it a good time to merge?

XVilka commented:
Ping? I addressed feedback multiple times I was asked to. Please either merge as is or just remove the old package completely - it's older than trilobites.

thesamesam commented:
This is on the proxy-maint proj's list. I can't merge it myself yet (no perms) but it is to be looked at. Please be patient, I understand you've waited a while, but we've had the ACK and will move forward when we can. It won't be anywhere near as long as you already waited.

juippis commented on profiles/updates/1Q-2020:
```diff
@@ -12,3 +12,4 @@ move www-servers/automx2 net-mail/automx2
 move dev-python/pysnmp-apps net-analyzer/snmpclitools
 move net-im/slack-bin net-im/slack
 slotmove sys-libs/libunwind 7 0
+move dev-lang/open-cobol dev-lang/gnucobol
```
This should be in 2Q-2020 file.

juippis commented on dev-lang/gnucobol/metadata.xml:
```diff
@@ -2,8 +2,12 @@
 <pkgmetadata>
 	<maintainer type="person">
-		<email>patrick@gentoo.org</email>
-		<name>Patrick Lauer</name>
```
Leave him also here, he's on devaway but might have personal interest towards this package when he returns.

juippis commented on dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:
```diff
@@ -0,0 +1,33 @@
+
+MY_P="${P/_/-}"
+
+DESCRIPTION="an open-source COBOL compiler"
```
`An`

juippis commented on dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:
```diff
@@ -0,0 +1,33 @@
+	sys-libs/ncurses:0"
+DEPEND="${RDEPEND}
+	berkdb? ( sys-libs/db:4.8 )
+	nls? ( sys-devel/gettext )"
```
This is a BDEPEND with EAPI-7.

juippis commented on dev-lang/gnucobol/gnucobol-3.0_rc1.ebuild:
```diff
@@ -0,0 +1,33 @@
+RDEPEND="dev-libs/gmp:0
+	sys-libs/ncurses:0"
+DEPEND="${RDEPEND}
+	berkdb? ( sys-libs/db:4.8 )
```
Are you sure it doesn't work with later ones? Upstream files don't indicate that anywhere. Also looking at my build.log it seems to be linking against 6.0 even though 4.8.30-r3 is installed. So do something like `>=sys-libs/db-4.1:*` here.

juippis commented:
The problem here is that the current maintainer is away, and has been for a long time. No one else sees notifications for these bugs / pull requests, and everyone else is pretty much busy with their own projects. Usually people are not that keen on touching packages belonging to others.

XVilka commented:
Fixed everything except GCC-10. I will check this one later, since the need to install it first. There wasn't any GCC-10 when I opened the pull request...

ajakk commented:
> Fixed everything except GCC-10. I will check this one later, since the need to install it first. There wasn't any GCC-10 when I opened the pull request...

Did you ever check into this?

XVilka commented:
@ajakk sorry, had no time on this and will not in the near future (I am busy with other FOSS projects).

XVilka commented:
If anyone is going to continue to work on this - please note that it's better to package 3.1.2 instead:

GnuCOBOL 3.1.2 — Latest release
Released 23 December 2020

GnuCOBOL 3.1.2 includes many new features compared to the previous release, while maintaining full source compatibility - (too much to note here but found in the linked NEWS), here are some of the highlights:

Huge improvements for compatibility to different COBOL dialects, better error handling and adjustable exceptions per COBOL 2002/2014; more modern format for diagnostic messages (especially useful when used in an integrated development environment possible in Emacs, Vim, VSCodium and others) and improved source-level debugging.

https://sourceforge.net/projects/gnucobol/files/gnucobol/3.1/gnucobol-3.1.2.tar.xz/download

atupone commented:
rename done

thesamesam commented:
Thank you @tupone and thank you @XVilka, I'm sorry this took so long.
